ckanext/xloader/jobs.py

```diff
@@ -541,8 +564,7 @@ def __init__(self, task_id, input):
         self.input = input
 
     def emit(self, record):
-        conn = db.ENGINE.connect()
-        try:
+        with db.ENGINE.connect() as conn:
             # Turn strings into unicode to stop SQLAlchemy
             # "Unicode type received non-unicode bind param value" warnings.
             message = str(record.getMessage())
```

Comment:
the problem we are trying to stop is not two workers picking up the same job (it might happen but very unlikely). Its the lock on the dataset when doing the meta update that usually hangs if another has a open transaction half way.

Reply:
That can be a future step.